lib/hooks/useValidateUser.ts

```diff
@@ -0,0 +1,17 @@
+import useSWR from "swr";
+
+interface UserResponse{
+  data: DbUser
+}
+const useValidateUser = (username:string)=>{
```

Comment:
The naming and the logic for this are off: the hook itself should test the `user.is_open_sauced_member` once it is fetched, while we don't have the data this is trying to do 2 things in one.

Please separate the data fetching patterns from checking the returned data itself: https://swr.vercel.app/docs/data-fetching